src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -139,54 +147,57 @@ private static List<DeploymentPackageConfiguration> buildDeploymentPackageConfig
     }
 
     /**
-     * Converts deployment configuration {@link Configuration} that is generated by CreateDeployment and gets sent down
-     * via IoT Job and shadow to the Nucleus's core {@link DeploymentDocument}.
+     * Converts deployment configuration {@link Configuration} that is generated by
+     * CreateDeployment and gets sent down via IoT Job and shadow to the Nucleus's
+     * core {@link DeploymentDocument}.
      *
-     * @param config Fleet configuration that is generated by CreateDeployment and gets sent down via IoT Job and
-     *               shadow
+     * @param config
+     *            Fleet configuration that is generated by CreateDeployment and gets
+     *            sent down via IoT Job and shadow
      * @return Nucleus's core {@link DeploymentDocument}
-     * @throws InvalidRequestException if failed to parsing deployment document from configuration.
+     * @throws InvalidRequestException
+     *             if failed to parsing deployment document from configuration.
      */
     public static DeploymentDocument convertFromDeploymentConfiguration(Configuration config)
             throws InvalidRequestException {
 
-        DeploymentDocument.DeploymentDocumentBuilder builder =
-                DeploymentDocument.builder().configurationArn(config.getConfigurationArn())
-                        .deploymentId(config.getDeploymentId())
-                        .requiredCapabilities(config.getRequiredCapabilities())
-                        .deploymentPackageConfigurationList(convertComponents(config.getComponents()))
-                        .groupName(parseGroupNameFromConfigurationArn(config))
-                        .timestamp(config.getCreationTimestamp());
+        DeploymentDocument.DeploymentDocumentBuilder builder = DeploymentDocument.builder()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against a allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html